maxfw/tests/test_image_utils.py

```diff
@@ -0,0 +1,171 @@
+
+
+def test_imageprocessor_normalize():
+    """Test the Imageprocessor's normalize function."""
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGBA'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGB'), Normalize(), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('L'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+
+def test_imageprocessor_standardize():
+    """Test the Imageprocessor's standardize function."""
+
+    # Test standardize
+    transform_sequence = [ToPILImage('RGBA'), Standardize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert round(np.std(img_out)) == 1
+
+    # Test standardize
+    transform_sequence = [ToPILImage('RGB'), Standardize(), Standardize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert round(np.std(img_out)) == 1
+
+    # Test standardize
+    transform_sequence = [ToPILImage('L'), Standardize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert round(np.std(img_out)) == 1
+
+
+def test_imageprocessor_rotate():
+    """Test the Imageprocessor's rotate function."""
+
+    # Test rotate (int)
+    transform_sequence = [ToPILImage('RGBA'), Resize((200, 200)), Rotate(5)]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+    # Test rotate (negative int)
+    transform_sequence = [ToPILImage('RGBA'), Resize((200, 200)), Rotate(-5)]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+    # Test rotate (float)
+    transform_sequence = [ToPILImage('RGBA'), Resize((200, 200)), Rotate(499.99)]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+
+def test_imageprocessor_combinations():
+    """Test various combinations of the Imageprocessor's functionality."""
+
+    # Combination 1
+    transform_sequence = [
+        ToPILImage('RGB'),
+        Resize((2000, 2000)),
+        Normalize(),
+        Rotate(5),
+        Grayscale(num_output_channels=4),
+        Resize((200, 200)),
+    ]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+
+def test_flask_error():
```

Comment:
These tests don't seem to be passing Travis. Not sure if that is due to things still being WIP or if there is something to fix here.

Reply:
Fixed in the next update. Thanks!